fix(coverage): Add files to be included or excluded during code coverage generation

nyakamwe opened the pull request:
**What does this PR do?**
It fixes bugs related to code coverage status generation.

**Description of Task to be completed?**
- adding all js files found in src to be included
- exclude files from the test folder

**How should this be manually tested?**
After cloning this repo, add or remove some codes and run: npm run coverage
this will deliver a report to the coveralls and generate coverage status in percentage.

**What are the relevant pivotal tracker stories?**
#182209989

[coveralls #182209989]

Ihimbazwe committed:
Testing heroku version 1

resolve conflicts

Ihimbazwe committed:
Added productuin mode to heroku

resolve conflicts

Ihimbazwe committed:
Added production mode to heroku

Ihimbazwe committed:
Added production mode to heroku

Ihimbazwe committed:
Added production mode to heroku

resolve conflicts

Ihimbazwe committed:
Added production mode to heroku

resolve conflicts

Ihimbazwe committed:
Added production mode to heroku

resolve conflicts

Ihimbazwe committed:
Added production mode to heroku

Ihimbazwe committed:
Added  Fixed babel config  dependencies  to heroku

resolve conflicts

Ihimbazwe committed:
fix(Conflicting files): removed the cors in index and packageloc

resolve conflicts

nyakamwe committed:
fix(coverage): Add files to be included or excluded

- adding all js file found in src to be included
- exclude files from test folder

[coveralls #182209989]

resolve conflicts

resolving

nyakamwe committed:
fix(coverage): Add files to be included or excluded

- adding all js file found in src to be included
- exclude files from test folder

[coveralls #182209989]

Himbaza123h commented:
Well done!

munezerobagira reviewed (commented):
Checking  on coveralls the `src/sum.js` is only file which  has the coverage ,  you can increase the coverage to other files as well
https://coveralls.io/builds/49892434
![image](https://user-images.githubusercontent.com/42154315/172916626-c87f43f3-492a-411f-b2a0-fbf33ab1e312.png)
 make sure that other files like src/routes/index.js are checked too

munezerobagira commented on package.json:
```diff
@@ -4,8 +4,8 @@
-    "test": "nyc --reporter=text --reporter=html mocha --recursive --require @babel/polyfill --require @babel/register src/test",
-    "coverage": "nyc  report --reporter=text-lcov --reporter=lcov",
+    "test": "mocha --recursive --require @babel/polyfill --require @babel/register ./src/test/*.js --timeout 30000 --exit",
+    "coverage": "nyc --reporter=text --reporter=lcov npm run test",
```
you can add the --all after nyc to collect  coverage for the entire project

munezerobagira replied:
Or you can add it in .nycrc.json

nyakamwe replied:
yes, I have added it in .nycrc.json

munezerobagira commented on .nycrc.json:
```diff
@@ -3,5 +3,8 @@
-    "lines": 80
+    "lines": 80,
+    "include": ["src/**/*.js"],
+    "exclude": ["src/**/*.test.js", "src/database/migrations/*.js"]
```
I am not sure if we will write the tests for seeders too, if possible you can add it  to exclude too

nyakamwe replied:
seeders are also excluded.

Ihimbazwe committed:
fix(missing): Added the Procfile for which was missing in this feat

K-Ds commented on .nycrc.json:
```diff
@@ -3,5 +3,8 @@
-    "lines": 80
+    "lines": 80,
+    "include": ["src/**/*.js"],
+    "exclude": ["src/**/*.test.js", "src/database/migrations/*.js"]
```
You can also add **"src/database/seeders/*.js** to the list

Ihimbazwe committed:
chore: modified the start script and procfile

resolve conflicts

nyakamwe committed:
fix(coverage): Do test for boost up coverage

- Test database connection for testing
- writing test for root and users routes

[Coverall #182209989]

nyakamwe committed:
fix(coverage): use env variables in tests

nyakamwe committed:
fix(coverage): Fixes server port

nyakamwe committed:
fix(coverage): Apply object destructuring for coverage boost

nyakamwe commented:
The threshold is above 80% in local but when it comes to CI it decreases unexpectedly

nyakamwe committed:
fix(coverage): Add before hook

- Adding connect to database in before hook

nyakamwe commented:
> Checking on coveralls the `src/sum.js` is only file which has the coverage , you can increase the coverage to other files as well https://coveralls.io/builds/49892434 ![image](https://user-images.githubusercontent.com/42154315/172916626-c87f43f3-492a-411f-b2a0-fbf33ab1e312.png) make sure that other files like src/routes/index.js are checked too
![coveralls](https://user-images.githubusercontent.com/53573955/173105220-7fb225fd-8744-4842-9ffc-3b8aac122fe0.png)

nyakamwe committed:
fix(coverage): change reporter to text-lcov

nyakamwe committed:
fix(coverage): change reporter to text-lcov

nyakamwe committed:
fix(coverage): update script for testing

nyakamwe committed:
fix(coverage): Set decrease for failure

nyakamwe committed:
fix(coverage): Set decrease for failure

Ihimbazwe committed:
chore: modified start script in package json for heroku deployment

resolve conflicts

istafrica-dennw reviewed (changes requested):
@nyakamwe ,
Your PR has conflicts

nyakamwe committed:
fix(coverage): resolve conflicts

resolving

nyakamwe committed:
fix(coverage): boost coverage

nyakamwe committed:
fix(coverage): fix url for database

nyakamwe committed:
fix(coverage): import dotenv in helper

nyakamwe committed:
fix(coverage): import dotenv in helper

nyakamwe committed:
fix(coverage): describe get default helper

nyakamwe committed:
fix(coverage): use cross-env for multi db

nyakamwe committed:
fix(coverage): use cross-env for multi db

nyakamwe committed:
fix(coverage): use cross-env for multi db

nyakamwe committed:
Merge branch 'develop' into fix-code-coverage

nyakamwe committed:
fix(coverage): add test to coverage

nyakamwe committed:
fix(coverage): add test to coverage
